lucene/core/src/java/org/apache/lucene/util/bkd/BKDReader.java

```diff
@@ -780,4 +783,43 @@ public int getDocCount() {
   public boolean isLeafNode(int nodeID) {
     return nodeID >= leafNodeOffset;
   }
+
+  protected static class BKDReaderDocIDSetIterator extends DocIdSetIterator {
+
+    int idx;
+    int length;
+    int offset;
+    int[] docIds;
+
+    public BKDReaderDocIDSetIterator(int maxPointsInLeafNode) {
+      this.docIds = new int[maxPointsInLeafNode];
+    }
+
+    @Override
+    public int docID() {
+      if (idx == -1)  {
+        return -1;
+      }
+      return docIds[offset + idx];
+    }
+
+    @Override
+    public int nextDoc() throws IOException {
+      if (idx == length - 1) {
+        return DocIdSetIterator.NO_MORE_DOCS;
+      }
+      idx++;
+      return docIds[offset + idx];
+    }
+
+    @Override
+    public int advance(int target) throws IOException {
+      throw new UnsupportedOperationException();
```

Comment:
I think it'd be better to support it, even if slow. Eg. you can return `slowAdvance(target)`.